Marlin/configuration_store.cpp

```diff
@@ -824,7 +824,13 @@ void Config_ResetDefault() {
     retract_recover_feedrate_mm_s = RETRACT_RECOVER_FEEDRATE;
   #endif
 
-  volumetric_enabled = false;
+  volumetric_enabled =
+  #if ENABLED(ENABLE_VOLUMETRIC_FILAMENT_DIAMETER)
+    true
+  #else
+    false
+  #endif
+  ;
   for (uint8_t q = 0; q < COUNT(filament_size); q++)
```

Comment:
The result of ENABLE()/DISABLE() is already boolean
```diff
@@ -822,11 +822,11 @@ void Config_ResetDefault() {
       retract_recover_length_swap = RETRACT_RECOVER_LENGTH_SWAP;
     #endif
     retract_recover_feedrate_mm_s = RETRACT_RECOVER_FEEDRATE;
   #endif
 
-  volumetric_enabled = false;
+  volumetric_enabled = ENABLED(ENABLE_VOLUMETRIC_FILAMENT_DIAMETER);
   for (uint8_t q = 0; q < COUNT(filament_size); q++)
     filament_size[q] = DEFAULT_NOMINAL_FILAMENT_DIA;
 
   endstops.enable_globally(
     #if ENABLED(ENDSTOPS_ALWAYS_ON_DEFAULT)
```
`The missing place to make it work when EEPROM is off is in Marlin_main.cpp:
```diff
@@ -386,11 +386,11 @@ float constexpr homing_feedrate_mm_s[] = {
 static float feedrate_mm_s = MMM_TO_MMS(1500.0), saved_feedrate_mm_s;
 int feedrate_percentage = 100, saved_feedrate_percentage,
     flow_percentage[EXTRUDERS] = ARRAY_BY_EXTRUDERS1(100);
 
 bool axis_relative_modes[] = AXIS_RELATIVE_MODES,
-     volumetric_enabled = false;
+     volumetric_enabled = ENABLED(ENABLE_VOLUMETRIC_FILAMENT_DIAMETER);
 float filament_size[EXTRUDERS] = ARRAY_BY_EXTRUDERS1(DEFAULT_NOMINAL_FILAMENT_DIA),
       volumetric_multiplier[EXTRUDERS] = ARRAY_BY_EXTRUDERS1(1.0);
 
 // The distance that XYZ has been offset by G92. Reset by G28.
 float position_shift[XYZ] = { 0 };
```